### DISCUSSION TITLE
Type safe usage of ConfigurationProperties in @Requires ?

### DISCUSSION THREAD

**Discussion by tobiasschaefer** [micronaut-projects/micronaut-core#4969] [Q&A] [answered]
I'm using `@ConfigurationProperties` a lot for type safe configuration classes with the default in a single place. That's perfect.

However, if I have a bean with a `@Required` annotation I need to
1. reference the property by name, i.e. a string
2. and duplicate the default value (if I already have a ConfigurationProperties class for it)

e.g.

```java
@Singleton
@Requires(property="foo", value="false", defaultValue="true") 
public class MyBean () {

}
```

Is there any way to reference the property of the `ConfigurationProperties`? This would make it type safe and also make sure that the default is only in the `ConfigurationProperties`.

Something like the following for a boolean:

```java
@Requires(value="#{myConfiguration.isFoo}")
```

Thanks!

**Comment by graemerocher** [comment, marked as answer]
Not currently no. One day we may implement an expression language, but not yet
